aten/src/ATen/native/AveragePool2d.cpp

```diff
@@ -8,59 +8,88 @@ namespace at {
 
   TORCH_CHECK(stride.empty() || stride.size() == 1 || stride.size() == 2,
     "avg_pool2d: stride must either be omitted, a single int, or a tuple of two ints");
-  const int dH = stride.empty() ? kH : safe_downcast<int, int64_t>(stride[0]);
-  const int dW = stride.empty() ? kW :
-                 stride.size() == 1 ? dH : safe_downcast<int, int64_t>(stride[1]);
+  int dH = stride.empty() ? kH : safe_downcast<int, int64_t>(stride[0]);
+  int dW = stride.empty() ? kW : stride.size() == 1 ? dH : safe_downcast<int, int64_t>(stride[1]);
 
   TORCH_CHECK(padding.size() == 1 || padding.size() == 2,
     "avg_pool2d: padding must either be a single int, or a tuple of two ints");
-  const int padH = safe_downcast<int, int64_t>(padding[0]);
-  const int padW = padding.size() == 1 ? padH : safe_downcast<int, int64_t>(padding[1]);
+  int padH = safe_downcast<int, int64_t>(padding[0]);
+  int padW = padding.size() == 1 ? padH : safe_downcast<int, int64_t>(padding[1]);
 
   TORCH_CHECK(!divisor_override.has_value() || divisor_override.value() != 0,
     "divisor must be not zero");
 
-  /* sizes */
-  const int64_t nbatch = input.ndimension() == 4 ? input.size(-4) : 1;
-  const int64_t nInputPlane = input.size(-3);
-  const int64_t inputHeight = input.size(-2);
-  const int64_t inputWidth = input.size(-1);
+  int64_t nbatch = input.ndimension() == 4 ? input.size(-4) : 1;
+  int64_t nInputPlane = input.size(-3);
+  int64_t inputHeight = input.size(-2);
+  int64_t inputWidth = input.size(-1);
 
-  const int64_t outputHeight = pooling_output_shape<int64_t>(inputHeight, kH, padH, dH, 1, ceil_mode);
-  const int64_t outputWidth = pooling_output_shape<int64_t>(inputWidth, kW, padW, dW, 1, ceil_mode);
+  int64_t outputHeight = pooling_output_shape<int64_t>(
+      inputHeight, kH, padH, dH, 1, ceil_mode);
+  int64_t outputWidth =
+      pooling_output_shape<int64_t>(inputWidth, kW, padW, dW, 1, ceil_mode);
 
   auto memory_format = input.suggest_memory_format();
   pool2d_shape_check(
-    input,
-    kH, kW, dH, dW, padH, padW, 1, 1,
-    nInputPlane,
-    inputHeight, inputWidth,
-    outputHeight, outputWidth, memory_format);
+      input,
+      kH,
+      kW,
+      dH,
+      dW,
+      padH,
+      padW,
+      1,
+      1,
+      nInputPlane,
+      inputHeight,
+      inputWidth,
+      outputHeight,
+      outputWidth,
+      memory_format);
 
   /* resize output */
   if (input.ndimension() == 3) {
-    set_output(0, {nInputPlane, outputHeight, outputWidth}, input.options());
+    set_output(
+        0,
+        {nInputPlane,
+         outputHeight,
+         outputWidth},
+        input.options());
   }
   else {
-    set_output(0, {nbatch, nInputPlane, outputHeight, outputWidth}, input.options().memory_format(memory_format));
+    set_output(
+        0,
+        {nbatch,
+         nInputPlane,
+         outputHeight,
+         outputWidth},
+        input.options().memory_format(memory_format));
   }
+
+  return {
+        kH=kH,
```

Comment:
huh, the dots are optional? Should we prefer if people specify dot or not?

Reply:
No, I think what's happening is that the `kH` declared above is being assigned to itself and then used to make the struct. The dots are definitely required. But on that note, I can't reproduce the error that is supposed to be emitted with this syntax: https://godbolt.org/z/Y6dxh1Eqd

I remember clearly that we did manage to write code that produces that error during a VC, but I can't seem to do it again. I tried gcc and clang both.

Reply:
blegh. Well I guess we'll just have to write a little lint rule for this. Maybe clang tidy can catch it.